File: 2188-minimized-maximum-of-products-distributed-to-any-store/minimized-maximum-of-products-distributed-to-any-store.py
class Solution:
    def minimizedMaximum(self, n: int, quantities: List[int]) -> int:
        mini,maxi=1,max(quantities)
        while mini<maxi:
            mid=mini+(maxi-mini)//2
            stores=sum([ceil(q/mid) for q in quantities])
            if stores<=n:
                maxi=mid
            else:
                mini=mid+1
        return maxi

File: 2188-minimized-maximum-of-products-distributed-to-any-store/minimized-maximum-of-products-distributed-to-any-store.cpp
class Solution {
public:
bool check(int mid, int n, vector<int> &quantities){
    int count = 0;
    for(int i=0;i<quantities.size();i++){
        if(quantities[i]%mid==0){
            count+=quantities[i]/mid;
        }
        else{
            count= count+(quantities[i]/mid)+1;
        }
    }
    if(count<=n) return true;
    return false;
}
    int minimizedMaximum(int n, vector<int>& quantities) {
        int low = 1;
        int high = *max_element(quantities.begin(), quantities.end());
        int res = -1;

        while(low<=high){
            int mid = low-(low-high)/2;
            if(check(mid, n, quantities)){
                res = mid;
                high = mid-1;
            }
            else{
                low = mid+1;
            }
        }
        return res;
    }
};